pkg/handlers/cluster.go

```diff
@@ -0,0 +1,57 @@
+package handlers
+
+import (
+	"net/http"
+
+	sdk "github.com/openshift-online/ocm-sdk-go"
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	log "github.com/sirupsen/logrus"
+)
+
+type ClusterHandler struct {
+	ocm       *sdk.Connection
+	clusterId string
+}
+
+// For /api/clusters_mgmt/v1/clusters/{cluster_id}
+// https://api.openshift.com/#/default/get_api_clusters_mgmt_v1_clusters__cluster_id_
+func NewClusterHandler(o *sdk.Connection, clusterId string) *ClusterHandler {
+	log.Debug("Creating new cluster object Handler")
+	return &ClusterHandler{
+		ocm:       o,
+		clusterId: clusterId,
+	}
+}
+
+// https://pkg.go.dev/github.com/openshift-online/ocm-sdk-go@v0.1.382/clustersmgmt/v1#Cluster
+func (g *ClusterHandler) GetCluster() (*cmv1.Cluster, error) {
+	log.Debugf("Sending get cluster object request to OCM API: %s", g.clusterId)
+	request := g.ocm.ClustersMgmt().V1().Clusters().Cluster(g.clusterId)
+	resp, err := request.Get().Send()
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body(), nil
+}
+
+// Proxies to
+// https://api.openshift.com/#/default/get_api_clusters_mgmt_v1_clusters__cluster_id_
+func (g *ClusterHandler) ServeClusterGet(w http.ResponseWriter, r *http.Request) {
+	switch r.Method {
+	case "GET":
```

Comment:
@feichashao - Thanks for the update, just an asking, do we anticipate the need to handle other HTTP methods in the future for something like `POST`/`PUT`/`DELETE` or this handler just strictly dealing with `GET` method? If it's the case,  it might be unnecessary to use `switch` statement ?

Reply:
Hey @xiaoyu74 , yes, in the current scope we will have `GET` only.  The `switch` is to follow the pattern of other existing handlers [example](https://github.com/openshift/ocm-agent/blob/cb233bb07656357b23cd16e05d00b6b2dc6d449c/pkg/handlers/upgrade_policies.go#L95).

Reply:
Yeah,  just thought it's not typical to use switch statement for only one condition. Looks I am too nitpicking, all good. :+1: